include/fiber_container_base.hpp

```diff
@@ -0,0 +1,165 @@
+#ifndef FIBER_CONTAINER_BASE_HPP
+#define FIBER_CONTAINER_BASE_HPP
+
+#include <kernels.hpp>
+#include <params.hpp>
+#include <skelly_sim.hpp>
+
+class Periphery;
+class BodyContainer;
+
+/// Abstract base class to represent a container holding fiber objects
+///
+/// This is done so that there is a common interface for how to interact with fiber objects that
+/// are local to each MPI rank.
+
+class FiberContainerBase {
+  public:
+    /// Enum for the different fiber types we know about
+    enum FIBERTYPE { None, FiniteDifference };
+
+    //! \name Constructors and destructors
+    //@{
+
+    /// Empty container constructor to avoid initializing list complications. No way to initialize
+    /// after using this constructor, so overwrite objects with full constructor.
+    FiberContainerBase() = default;
+    /// Constructor using TOML and params
+    FiberContainerBase(toml::array &fiber_tables, Params &params);
+
+    //@}
+
+    //! \name Public member functions
+    //@{
+
+    /// @brief Set the pair interaction evaluator
+    void set_evaluator(const std::string &evaluator);
+
+    /// @brief Get the local number of fibers in the container
+    int get_local_fiber_number() const { return n_local_fibers_; }
+
+    /// @brief Get the local number of nodes across fibers in the container
+    int get_local_node_count() const { return n_local_node_count_; }
+
+    /// @brief Get the local solution size across fibers in this container
+    int get_local_solution_size() const { return n_local_solution_size_; }
+
+    /// @brief Get the global number of fibers in all MPI ranks
+    int get_global_fiber_number() const;
+
+    /// @brief Get the global number of nodes for fibers in all MPI ranks
+    int get_global_node_count() const;
+
+    /// @brief Get the global solution size across fibers in all MPI ranks
+    int get_global_solution_size() const;
+
+    /// @brief Set the local fiber number, node count, and solution size
+    void set_local_fiber_numbers(int n_fibers, int n_local_nodes, int n_local_solution_size);
```

Comment:
This is a powerful footgun. Can we get rid of it? Since the `fc` has all its fibers, (even if some are deactivated), it should be able to update these numbers without an external setter. Something like `update_internal_counts()`